app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,41 @@
+class MerchantsController < ApplicationController
+
+  def index
+    @merchants = Merchant.all
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def new
+  end
+
+  def create
+    @merchant = Merchant.create(merchant_params)
+    
+    redirect_to "/merchants"
+  end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.update(merchant_params)
+    @merchant = Merchant.find(params[:id])
```

Comment:
Rather than calling `Merchant.update`, you should call update on a specific merchant. Capital M Merchant is the class which represents *all* the merchants. You want to update a single merchant. `Merchant.update` will update the last merchant in your database, so your tests still pass. Here is an example of how you could have done it:

```
@merchant = Merchant.find(params[:id])
@merchant.update(merchant_params)

```